spread.yaml

```diff
@@ -0,0 +1,122 @@
+project: core-initrd
+
+environment:
+  SETUPDIR: /home/core-initrd
+  PROJECT_PATH: $SETUPDIR
+  PATH: $PATH:$PROJECT_PATH/tests/bin
+  TESTSLIB: $PROJECT_PATH/tests/lib
+  TESTSTOOLS: $PROJECT_PATH/tests/lib/tools
+  # TODO: are these vars needed still?
+  LANG: "C.UTF-8"
+  LANGUAGE: "en"
+
+backends:
+  google-nested:
+    type: google
+    key: '$(HOST: echo "$SPREAD_GOOGLE_KEY")'
+    location: computeengine/us-east1-b
+    plan: n1-standard-2
+    halt-timeout: 2h
+    systems:
+      - ubuntu-20.04-64:
+          image: ubuntu-2004-64-virt-enabled
+          storage: 20G
+
+  qemu-nested:
+    type: qemu
+    # TODO:UC20: uc20 needs 2G or grub will not loopback the kernel snap
+    #            we use nested VM's, so use 4G for the host until we can boot
+    #            uc20 with less RAM
+    memory: 4G
+    systems:
+      - ubuntu-20.04-64:
+          username: ubuntu
+          password: ubuntu
+
+path: /home/core-initrd
+
+exclude:
+  - .git
+  - "*.o"
+  - "*.a"
+  - "*.snap"
+  - "*.img"
+
+prepare: |
+  # NOTE: This part of the code needs to be in spread.yaml as it runs before
+  # the rest of the source code (including the tests/lib directory) is
+  # around. The purpose of this code is to fix some connectivity issues setup 
+  # spread functions to be more usable outside of the explicit task.yaml 
+  # sections.
+
+  # apt update is hanging on security.ubuntu.com with IPv6, prefer IPv4 over IPv6
```

Comment:
huh, no way. Is there a bug about this? Are you sure that nested VMs have correct ipv6 passthrough setup?

Reply:
well this was originally added to snapd's spread.yaml as of https://github.com/snapcore/snapd/pull/3418, so perhaps things are better now, I will leave it to you to look into removing this, if it works for you to remove it, then let us know and we'd be happy to remove the same from our spread.yaml :-)